packages/cache/src/cache.ts

```diff
@@ -85,17 +87,38 @@ export async function restoreCache(
     checkKey(key)
   }
 
-  const compressionMethod = await utils.getCompressionMethod()
+  let cacheEntry: ArtifactCacheEntry | null
+  let compressionMethod = await utils.getCompressionMethod()
   let archivePath = ''
   try {
-    // path are needed to compute version
-    const cacheEntry = await cacheHttpClient.getCacheEntry(keys, paths, {
-      compressionMethod
-    })
-
-    if (!cacheEntry?.archiveLocation) {
-      // Cache not found
-      return undefined
+    try {
+      // path are needed to compute version
+      cacheEntry = await cacheHttpClient.getCacheEntry(keys, paths, {
+        compressionMethod
+      })
+
+      if (!cacheEntry?.archiveLocation) {
+        // Cache not found
+        return undefined
+      }
+    } catch (error) {
+      if (
+        process.platform === 'win32' &&
+        compressionMethod !== CompressionMethod.Gzip
+      ) {
+        // On windows, we will try to download the cache entry with the same key
+        // but with different compression method. This is to support the old cache entry created
+        // by the old version of the cache action.
+        compressionMethod = CompressionMethod.Gzip
+        cacheEntry = await cacheHttpClient.getCacheEntry(keys, paths, {
+          compressionMethod
+        })
```

Comment:
This can be moved to a well named function for the specific case for ease of understanding, and not take too much space in the overall function.

```suggestion
      cacheEntry = await getCacheEntryOnWindowsForFallbackCompression(keys, paths, compressionMethod);
```